pkg/proxy/proxy.go

```diff
@@ -177,16 +179,34 @@ func (p *proxy) invokeGrpc(writer http.ResponseWriter, request *http.Request) {
 	}
 }
 
-func writeError(writer http.ResponseWriter, err error) {
+type invocationError struct {
+	Error string `json:"error"`
+}
+
+func writeError(writer http.ResponseWriter, err error, accept string) {
+	textPos := strings.Index(accept, "text/plain")
+	jsonPos := strings.Index(accept, "application/json")
+	preferJSON := jsonPos > -1 && (textPos == -1 || jsonPos < textPos)
```

Comment:
the order of a key in the accept header doesn't mean much. I do like using a substring match instead of a literal match for application/json.